activerecord/test/models/admin/region.rb

```diff
@@ -0,0 +1,6 @@
+# The class Region (without namespace) shouldn't exist
+# in order to the test test_type_mismatch_with_namespaced_class
+# be valid
+class Admin::Region < ActiveRecord::Base
```

Comment:
Do we need to create new global models for a single test-case? We generally try to avoid extending those further. Would be nice if we can reproduce the problem with models defined in the test-case itself.